cmd/ci-secret-bootstrap/main.go

```diff
@@ -565,39 +570,35 @@ func updateSecrets(secretsGetters map[string]coreclientset.SecretsGetter, secret
 	return utilerrors.NewAggregate(errs)
 }
 
-func writeSecrets(secretsMap map[string][]*coreapi.Secret) error {
-	var tmpFiles []*os.File
-	defer func() {
-		for _, tf := range tmpFiles {
-			tf.Close()
-		}
-	}()
+func writeSecrets(secretsMap map[string][]*coreapi.Secret, dryLogOutput string) error {
+	objects := make(map[string][]kuberuntime.Object)
 
 	for cluster, secrets := range secretsMap {
-		tmpFile, err := ioutil.TempFile("", fmt.Sprintf("%s_*.yaml", cluster))
-		if err != nil {
-			return fmt.Errorf("failed to create tempfile: %w", err)
-		}
-		tmpFiles = append(tmpFiles, tmpFile)
-
-		logrus.Infof("Writing secrets from cluster %s to %s", cluster, tmpFile.Name())
-		if err := writeSecretsToFile(secrets, tmpFile); err != nil {
-			return fmt.Errorf("error while writing secrets for cluster %s to file %s: %w", cluster, tmpFile.Name(), err)
+		var s []kuberuntime.Object
+		for _, secret := range secrets {
+			s = append(s, secret.DeepCopyObject())
 		}
+		objects[cluster] = s
 	}
-	return nil
-}
 
-func writeSecretsToFile(secrets []*coreapi.Secret, w io.Writer) error {
-	serializerOptions := kubejson.SerializerOptions{Yaml: true, Pretty: true, Strict: true}
-	serializer := kubejson.NewSerializerWithOptions(kubejson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme, serializerOptions)
+	data, err := yaml.Marshal(objects)
```

Comment:
If we are serializing objects, we must use the correct scheme. This change is incorrect.

Reply:
We don't serialize anything now, so the current code wasn't needed. That is why I changed it. Either way. I will close this PR and work on the integration tests. Thanks a lot for the explanation!

Reply:
/close